documentation/src/main/asciidoc/topics/search.adoc

```diff
@@ -0,0 +1,1096 @@
+This configuration includes which fields are annotated, which analyzers should be used, how to map nested objects and so on.
+Detailed documentation is available at link:https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#search-mapping[the Hibernate Search manual].
+
+==== @DocumentId
+Unlike Hibernate Search, using _@DocumentId_ to mark a field as identifier does not apply to {brandname} values; in {brandname} the identifier for all _@Indexed_ objects is the key used to store the value. You can still customize how the key is indexed using a combination of _@Transformable_ , custom types and custom _FieldBridge_ implementations.
+
+==== @Transformable keys
+The key for each value needs to be indexed as well, and the key instance must be transformed in a _String_. {brandname} includes some default transformation routines to encode common primitives, but to use a custom key you must provide an implementation of _org.infinispan.query.Transformer_ .
+
+[small]*Registering a key Transformer via annotations*
+
+You can annotate your key class with _org.infinispan.query.Transformable_ and your custom transformer implementation
+will be picked up automatically:
+
+[source,java]
+----
+
+@Transformable(transformer = CustomTransformer.class)
+public class CustomKey {
+   ...
+}
+
+public class CustomTransformer implements Transformer {
+   @Override
+   public Object fromString(String s) {
+      ...
+      return new CustomKey(...);
+   }
+
+   @Override
+   public String toString(Object customType) {
+      CustomKey ck = (CustomKey) customType;
+      return ...
+   }
+}
+
+----
+
+[small]*Registering a key Transformer via the cache indexing configuration*
+
+You can use the _key-transformers_ xml element in both embedded and server config:
+
+[source,xml,options="nowrap",subs=attributes+]
+----
+include::config_examples/indexing_key_transformer.xml[]
+----
+
+or alternatively, you can achieve the same effect by using the Java configuration API (embedded mode):
+
+[source,java]
+----
+
+   ConfigurationBuilder builder = ...
+   builder.indexing().autoConfig(true)
+         .addKeyTransformer(CustomKey.class, CustomTransformer.class);
+
+----
+
+[small]*Registering a Transformer programmatically at runtime*
+
+Using this technique, you don't have to annotate your custom key type and you also do not add the transformer to the,
+cache indexing configuration, instead, you can add it to the _SearchManagerImplementor_ dynamically at runtime by invoking
+_org.infinispan.query.spi.SearchManagerImplementor.registerKeyTransformer(Class<?>, Class<? extends Transformer>)_:
+
+[source,java]
+----
+org.infinispan.query.spi.SearchManagerImplementor manager = Search.getSearchManager(cache).unwrap(SearchManagerImplementor.class);
+manager.registerKeyTransformer(keyClass, keyTransformerClass);
+----
+
+NOTE: This approach is deprecated since 10.0 because it can lead to situations when a newly started node receives cache
```

Comment:
I'd leave out the "since 10.0" because it doesn't work downstream. Also users probably don't care about the since so much as the fact it is deprecated.

Reply:
This is basically the 10.0 docs, I forgot to rework this section removing deprecated code